components/script/dom/document.rs

```diff
@@ -1563,45 +1563,10 @@ impl Document {
 
     pub fn finish_load(&self, load: LoadType) {
         debug!("Document got finish_load: {:?}", load);
-        // The parser might need the loader, so restrict the lifetime of the borrow.
-        {
-            let mut loader = self.loader.borrow_mut();
-            loader.finish_load(&load);
-        }
-
-        if let LoadType::Script(_) = load {
-            self.process_deferred_scripts();
-            self.process_asap_scripts();
-        }
-
-        if let Some(parser) = self.get_current_parser() {
-            let ready_to_be_executed = match self.pending_parsing_blocking_script.borrow_mut().as_mut() {
-                Some(pending) => {
-                    if self.script_blocking_stylesheets_count.get() > 0 {
-                        return;
-                    }
-                    if let Some(pair) = pending.take_result() {
-                        Some(pair)
-                    } else {
-                        return;
-                    }
-                },
-                None => None,
-            };
-            if let Some((element, result)) = ready_to_be_executed {
-                *self.pending_parsing_blocking_script.borrow_mut() = None;
-                parser.resume_with_pending_parsing_blocking_script(&element, result);
-            }
-        } else if self.reflow_timeout.get().is_none() {
-            // If we don't have a parser, and the reflow timer has been reset, explicitly
-            // trigger a reflow.
-            if let LoadType::Stylesheet(_) = load {
-                self.window.reflow(ReflowGoal::ForDisplay,
-                                   ReflowQueryType::NoQuery,
-                                   ReflowReason::StylesheetLoaded);
-            }
-        }
+        self.loader.borrow_mut().finish_load(&load);
```

Comment:
No. Only `LoadBlocker::terminate` and the media load type are trivially handled. `stylesheet_loader` has very special behaviour, the parser too, and scripts too.